scripts/host_cap.py

```diff
@@ -352,10 +366,23 @@ def add_constituent_vars(cap, host_model, suite_list, run_env):
                 vtype = cvar.get_prop_value('type')
                 vkind = cvar.get_prop_value('kind')
                 ddt_mdata.append(f" type = {vtype} | kind = {vkind}")
-                const_stdnames.add(std_name)
+                if is_tend_var:
+                    tend_vars.add(cvar)
+                    tend_stdnames.add(std_name)
+                else:
+                    const_vars.add(cvar)
+                    const_stdnames.add(std_name)
+                # end if
+
             # end if
         # end for
     # end for
+    # Check that all tendency variables are valid
+    errmsg, errflg = check_tendency_variables(tend_vars, const_vars)
```

Comment:
Most of this code/logic in check_tendency_variables already exists in the the [VarCompatObj](https://github.com/NCAR/ccpp-framework/blob/main/scripts/var_props.py) ([and here](https://github.com/NCAR/ccpp-framework/blob/main/scripts/metavar.py#L375)).
Could the ability to check for "compatibility between any variable, not just a const variable, and its tendency" be folded into there? (This would make it more useable if someone wanted to do something similar with "state_variables" in the _pre/_post scheme calls.)
I'd be happy to help.

Reply:
I took a stab at modifying `compatible` to validate tendencies. Let me know what you think!